webapp/tests/test_functions.py

```diff
@@ -866,6 +866,38 @@ def test_consolidateBy(self):
             results = functions.consolidateBy({}, seriesList, func)
             self._verify_series_consolidationFunc(results, func)
 
+    def test_smooth(self):
+        seriesList = [
+            TimeSeries('collectd.test-db1.load.value',0,1,1,range(20)),
+        ]
+
+        def mock_evaluateTokens(reqCtx, tokens, replacements=None):
+            seriesList = [
+                TimeSeries('collectd.test-db1.load.value',0,1,1,range(20)),
+            ]
+            for series in seriesList:
+                series.pathExpression = series.name
+            return seriesList
+
+        expectedResults = [
+            TimeSeries('smooth(collectd.test-db1.load.value, 5)',0,1,1,[None,None,None,None,2,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20])
+        ]
+
+        with patch('graphite.render.functions.evaluateTokens', mock_evaluateTokens):
+            result = functions.smooth(
+                {
+                    'template': {},
+                    'args': ({},{}),
+                    'startTime': datetime(1970, 1, 1, 0, 0, 0, 0, pytz.timezone(settings.TIME_ZONE)),
+                    'endTime': datetime(1970, 1, 1, 0, 9, 0, 0, pytz.timezone(settings.TIME_ZONE)),
+                    'localOnly': False,
+                    'width': 400,
+                    'data': []
+                },
+                seriesList, 5
+            )
```

Comment:
For anyone following along that is curious how I debug/write tests, I add the following code (or something similar) after the execution of the function to see what is being returned vs what was coded as expected:
```
        for index, series in enumerate(result):
          print series.getInfo()
          print expectedResults[index].getInfo()
```
This gets removed before checking in.  Running 

It may be worthwhile to make TimeSeries.__repr__ spit out more data than it currently does.